ethexe/contracts/src/Router.sol

```diff
@@ -207,6 +229,46 @@ contract Router is IRouter, OwnableUpgradeable, ReentrancyGuardTransient {
     }
 
     // # Validators calls.
+
+    /// @dev Set validators for the next era.
+    function commitValidators(Gear.ValidatorsCommitment calldata commitment, bytes[] calldata signatures) external {
+        Storage storage router = _router();
+
+        uint256 currentEraIndex = (block.timestamp - router.genesisBlock.timestamp) / router.durations.era;
+
+        require(commitment.eraIndex == currentEraIndex + 1, "commitment era index is not next era index");
+
+        uint256 nextEraStart = router.genesisBlock.timestamp + router.durations.era * commitment.eraIndex;
+        require(block.timestamp >= nextEraStart - router.durations.election, "election is not yet started");
+
+        bool useValidators0 = Gear.currentEraValidatorsStoredInValidators1(router);
```

Comment:
use here `Gear.currentEraValidators` instead of `bool useValidators0`. it may simplify code

Reply:
As you can see I must use NOT current era validators

Reply:
I mean that it could be rewritten like this: (to avoid code duplication)
```solidity
Gear.Validators storage currentEraValidators = Gear.currentEraValidators(router);
require(
    currentEraValidators.useFromTimestamp < block.timestamp,
    "looks like validators for next era are already set"
);

bytes32 commitmentHash = ...

_resetValidators(currentEraValidators, commitment.validators, nextEraStart);
```

Reply:
this would also require introducing method opposite `Gear.currentEraValidators()`

Reply:
Ok, make cense if you mean appending something like `validatorsFreeSlot()`

Reply:
done